src/App.tsx

```diff
@@ -1,26 +1,163 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useRef, useState } from 'react';
+// import { UserWarning } from './UserWarning';
+import { addTodo, deleteTodo, getTodos, USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import { FilterBy } from './types/FilterBy';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { Errors } from './types/Errors';
+import { ErrorNotification } from './components/ErrorNotification';
 
-const USER_ID = 0;
+function filterTodos(todos: Todo[], show: string) {
+  if (show === FilterBy.All) {
+    return todos;
+  }
 
-export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+  switch (show) {
+    case FilterBy.Active: {
+      return todos.filter(todo => !todo.completed);
+    }
+
+    case FilterBy.Completed: {
+      return todos.filter(todo => todo.completed);
+    }
+
+    default:
+      return todos;
   }
+}
+
+export const App: React.FC = () => {
+  // if (!USER_ID) {
+  //   return <UserWarning />;
+  // }
+
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState(Errors.Default);
+  const [filterBy, setFilterBy] = useState(FilterBy.All);
+  const [isLoadedIDs, setIsLoadedIDs] = useState<number[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const timerId = useRef(0);
+
+  const changeErrorMesssage = (newErrorMessage: Errors) => {
+    window.clearTimeout(timerId.current);
+
+    setErrorMessage(newErrorMessage);
+    timerId.current = window.setTimeout(
+      () => setErrorMessage(Errors.Default),
+      3000,
+    );
+  };
+
```

Comment:
very heavy + this function breaks the SOLID principles. S - single responsibility
Function should set state or work with timer
Let's move timer in useffect in error component